src/genkey/clu_genkey.c

```diff
@@ -146,140 +149,273 @@ int wolfCLU_genKey_ED25519(WC_RNG* rng, char* fOutNm, int directive, int format)
 }
 #endif /* HAVE_ED25519 */
 
+#ifdef HAVE_ECC
+/* returns 0 on successfully writing out public DER key */
+static int wolfCLU_ECC_write_pub_der(WOLFSSL_BIO* out, WOLFSSL_EC_KEY* key)
+{
+    int derSz, ret = 0;
+    unsigned char *der = NULL;
+
+    if (out == NULL || key == NULL)
+        return BAD_FUNC_ARG;
+
+    derSz = wc_EccPublicKeyDerSize(key->internal, 1);
+    if (derSz <= 0) {
+        printf("error getting der size\n");
+        ret = derSz;
+    }
+
+    if (ret == 0) {
+        der = (byte*)XMALLOC(derSz, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
+        if (der == NULL) {
+            ret = MEMORY_E;
+        }
+        else {
+            derSz = wc_EccPublicKeyToDer(key->internal, der, derSz, 1);
+            if (derSz < 0) {
+                ret = derSz;
+            }
+        }
+    }
+
+    if (ret == 0) {
+        ret = wolfSSL_BIO_write(out, der, derSz);
+        if (ret != derSz) {
+            ret = -1;
+        }
+    }
+
+    if (der != NULL)
+        XFREE(der, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
+
+    return ret;
+}
+
+
+/* returns 0 on successfully writing out private DER key */
+static int wolfCLU_ECC_write_priv_der(WOLFSSL_BIO* out, WOLFSSL_EC_KEY* key)
```

Comment:
Not important for this merge, but this and the function above could probably be rolled into one. Then, just add a parameter to indicate if you want private or public.